src/upgrader.js

```diff
@@ -186,7 +186,7 @@ class Upgrader {
           const { stream, protocol } = await mss.handle(Array.from(this.protocols.keys()))
           log('%s: incoming stream opened on %s', direction, protocol)
           connection.addStream(stream, protocol)
-          this._onStream({ connection, stream, protocol })
+          this._onStream({ connection, stream, protocol, remotePeer })
```

Comment:
There shouldn't be a need to pass the `remotePeer`, as it's on the connection, `connection.remotePeer`. Same with the other two changes below